lib/internal/crypto/random.js

```diff
@@ -119,6 +120,51 @@ function randomFill(buf, offset, size, cb) {
   _randomBytes(buf, offset, size, wrap);
 }
 
+// Largest integer we can read from a buffer.
+// e.g.: Buffer.from("ff".repeat(6), "hex").readUIntBE(0, 6);
+const RAND_MAX = 281474976710655;
+
+function randomInt(min, max, cb) {
+  if (typeof cb !== 'function') {
+    throw new ERR_INVALID_CALLBACK(cb);
+  }
+  if (!Number.isInteger(min)) {
+    throw new ERR_INVALID_ARG_TYPE('min', 'integer', min);
+  }
+  if (!Number.isInteger(max)) {
+    throw new ERR_INVALID_ARG_TYPE('max', 'integer', min);
+  }
+  if (!(max >= min)) {
+    throw new ERR_OUT_OF_RANGE('max', `>= ${min}`, max);
+  }
+  if (!(max <= RAND_MAX)) {
+    throw new ERR_OUT_OF_RANGE('max', `<= ${RAND_MAX}`, max);
+  }
+
+  // First we generate a random int between [0..bound)
+  const bound = max - min;
+  const range = bound - 1;
+  const excess = RAND_MAX % range;
+  const randLimit = RAND_MAX - excess;
+
+  const pickAttempt = () => {
+    randomBytes(6, (err, bytes) => {
+      if (err) return cb(err);
+      const x = bytes.readUIntBE(0, 6);
+      // If x > (maxVal - (maxVal % range)), we will get "modulo bias"
+      if (x > randLimit) {
+        // Try again
+        return pickAttempt();
+      }
+
+      const n = (x % range) + min;
+      cb(null, n + min);
```

Comment:
`(x % range) + min + min` is confusing, or incorrect.

Reply:
Oops, didn't expect this to get reviewed so quickly! I had a hard time running the tests on my slow laptop. Fixed in latest push.